webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/reports/ExecutionBatch.scala

```diff
@@ -924,6 +924,27 @@ object ExecutionBatch extends Loggable {
 
     TimingDebugLogger.trace(s"Compliance: mergeCompareByRule - compute buildRuleNodeStatusReport: ${t11-t10}ms")
 
+    val (computed, newStatus) = currentRunReports.foldLeft((List[RuleNodeStatusReport](), List[RuleNodeStatusReport]())) { case ( (c,n), currentStatusReports) =>
+      complianceForRun.get(currentStatusReports.ruleId) match {
+        case None => //the whole rule is new!
+          //here, the reports are ACTUALLY pending, not missing.
+          (c,  currentStatusReports :: n )
+
+        case Some(runStatusReport) => //look for added / removed directive
+          val runDirectives = runStatusReport.directives
+          val currentDirectives = currentStatusReports.directives
+
+          //don't keep directive that were removed between the two configs
+          val toKeep = runDirectives.filterKeys(k => currentDirectives.keySet.contains(k))
+
+          //now override currentDirective with the one to keep in currentReport
+          val updatedDirectives = currentDirectives ++ toKeep
+          val newCompliance = runStatusReport.copy(directives = updatedDirectives)
+
+          (newCompliance :: c, n)
```

Comment:
i'm a bit unsure about this change: I prepend in a list, so it should be better.

Reply:
you can't be better than prepending a list, so it seems safe